packages/desktop/components/popup/popups/SignEvmTransactionPopup.svelte

```diff
@@ -81,11 +84,16 @@
     async function sign(): Promise<void> {
         isBusy = true
         try {
-            const result = await signEvmTransaction(transaction, chainId, account)
+            let result: string | TransactionReceipt = await signEvmTransaction(transaction, chainId, account)
+
+            if (signAndSend) {
+                const provider = chain.getProvider()
+                result = await provider?.eth.sendSignedTransaction(result)
+            }
```

Comment:
could you reuse the `signAndSendEvmTransaction` , maybe with a flag for this here? it includes the same logic + sets the loading state

Reply:
as well we'd need to include the rest of the code you can see in `sendTransactionFromEvm` e.g. persistingEvmTransactions